src/comms.cpp

```diff
@@ -1633,22 +1700,38 @@ void comms_loop()
 /**************************** CONTROLLER CODE *******************************
  * SECURITY+1.0
  */
+// TRANSMIT SEC+1.0 byte
+// PERF: takes aprox 14ms (including delay(5))
 bool transmitSec1(byte toSend)
 {
     bool noSend = false;
 
-    // safety
+#ifdef SEC1_COMMS_PERF_TIMES
+    _millis_t txbegin;
+    txbegin = _millis();
+#endif
+
+    // safety #1
     if (sw_serial.available())
     {
         ESP_LOGD(TAG, "SEC1 TX incoming data detected, cannot send right now");
         noSend = true;
     }
-
+    // safety #2
     if (digitalRead(UART_RX_PIN))
     {
         ESP_LOGD(TAG, "SEC1 TX UART_RX_PIN HIGH detected, cannot send right now");
         noSend = true;
     }
+    // safety #3
+    bool isRxPending = rxPending.load();
+    if (isRxPending)
```

Comment:
Similar... could this be `else if (rxPending.load())` as there is no need to assign to a variable (unless used later) and no need to test if `noSend` already true.